src/System.Reflection.MetadataLoadContext/src/System/Reflection/Runtime/BindingFlagSupport/QueriedMemberList.cs

```diff
@@ -0,0 +1,208 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Reflection.Core.Execution;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Reflection.Runtime.TypeInfos;
+using RuntimeTypeInfo = System.Reflection.TypeLoading.RoType;
+
+namespace System.Reflection.Runtime.BindingFlagSupport
+{
+    //
+    // Stores the result of a member filtering that's filtered by name and visibility from base class (as defined by the Type.Get*() family of apis).
+    //
+    // The results are as if you'd passed in a bindingFlags value of "Public | NonPublic | Instance | Static | FlattenHierarchy"
+    // In addition, if "ignoreCase" was passed to Create(), BindingFlags.IgnoreCase is also in effect.
+    //
+    // Results are sorted by declaring type. The members declared by the most derived type appear first, then those declared by his base class, and so on.
+    // The Disambiguation logic takes advantage of this.
+    //
+    // This object is a good candidate for long term caching.
+    //
+    // QueryMemberList's come in two flavors: ImmediateTypeOnly and full. The immediateTypeOnly only holds the results for one type, not any of its
+    // base types. This is used when the binding flags passed to a Get() api limit the search to the immediate type only in order to avoid triggering
+    // unnecessary assembly resolving and a lot of unnecessary ParameterInfo creation and comparison checks.
+    //
+    internal sealed class QueriedMemberList<M> where M : MemberInfo
```

Comment:
nit: `TMemberInfo` instead of M?

Reply:
Yes changing that would better adhere, but this code is also duplicated on corect, so I'd prefer to keep it consistent for now for easier diffing.